arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Validated.kt

```diff
@@ -364,7 +364,71 @@ public typealias Invalid<E> = Validated.Invalid<E>
  * ## Sequential Validation
  *
  * If you do want error accumulation, but occasionally run into places where sequential validation is needed,
- * then Validated provides a `withEither` method to allow you to temporarily turn a Validated
+ * then Validated provides a couple of methods that can be used.
+ *
+ * ### `andThen`
+ *
+ * The `andThen` method is similar to `flatMap`. In case of a valid instance it will pass the valid value into
+ * the supplied function that in turn returns a `Validated` instance
+ *
+ * ```kotlin:ank:playground
+ * import arrow.core.Validated
+ * import arrow.core.computations.either
+ * import arrow.core.valid
+ * import arrow.core.invalid
+ *
+ * abstract class Read<A> {
+ *  abstract fun read(s: String): A?
+ *
+ *  companion object {
+ *
+ *   val stringRead: Read<String> =
+ *    object : Read<String>() {
+ *     override fun read(s: String): String? = s
+ *    }
+ *
+ *   val intRead: Read<Int> =
+ *    object : Read<Int>() {
+ *     override fun read(s: String): Int? =
+ *      if (s.matches(Regex("-?[0-9]+"))) s.toInt() else null
+ *    }
+ *  }
+ * }
+ *
+ * data class Config(val map: Map<String, String>) {
+ *   suspend fun <A> parse(read: Read<A>, key: String) = either<ConfigError, A> {
+ *     val value = Validated.fromNullable(map[key]) {
+ *       ConfigError.MissingConfig(key)
+ *     }.bind()
+ *     val readVal = Validated.fromNullable(read.read(value)) {
+ *       ConfigError.ParseConfig(key)
+ *     }.bind()
+ *     readVal
+ *   }.toValidatedNel()
+ * }
+ *
+ * sealed class ConfigError {
+ *  data class MissingConfig(val field: String) : ConfigError()
+ *  data class ParseConfig(val field: String) : ConfigError()
+ * }
+ *
+ * //sampleStart
+ * val config = Config(mapOf("house_number" to "-42"))
+ *
+ * suspend fun main() {
+ *   val houseNumber = config.parse(Read.intRead, "house_number").andThen { number ->
+ *     if (number >= 0) Valid(0)
```

Comment:
@magott should it rather be `Valid(number)` here?

Reply:
🕵️

Reply:
haha, surprised there weren't more errors in code written in doc. Do you want a new PR @nomisRev ? Or will you just fix it in main?

Reply:
Looks great! Good job 👏 Looking forward to delete the extension method from our code ❤️

Reply:
I'll fix it directly in main.

This is actually why I'm working on this Dokka Plugin: https://github.com/nomisRev/AnkDokkaPlugin/pull/7/files
I think we can already do it today too with our current setup, but if we wrote tests instead of examples than we could catch these mistakes on CI as well.

Reply:
Yeah, I think the docs compile as part of the build, so this isn't an error as such. Just a logical bug.